src/ir/selector_test.cc

```diff
@@ -0,0 +1,94 @@
+#include "src/ir/selector.h"
+
+#include "absl/hash/hash_testing.h"
+#include "src/common/testing/gtest.h"
+
+namespace raksha::ir {
+
+// An enumeration of the different specific types of selectors that may
+// appear in a Selector.
+enum SelectorKind {
```

Comment:
Probably. The reason I did this was because I wanted to express my intention of the specific type independent of the bookkeeping `absl::variant` does in the implementation. That way, if I accidentally constructed the wrong specific type in an implementation (for instance) I wouldn't get an incorrect passing result.